src/analyz/ty_json.rs

```diff
@@ -857,44 +928,285 @@ fn render_constant<'tcx>(
+                ),
+            ).into())
+        }
+
+        fn call_intrinsic(
+            ecx: &mut InterpCx<'mir, 'tcx, Self>,
+            span: Span,
+            instance: Instance<'tcx>,
+            args: &[OpTy<'tcx, Self::PointerTag>],
+            ret: Option<(PlaceTy<'tcx, Self::PointerTag>, BasicBlock)>,
+            unwind: Option<BasicBlock>
+        ) -> InterpResult<'tcx> {
+            Err(InterpError::Unsupported(
+                UnsupportedOpInfo::Unsupported(
+                    "call_intrinsic".into(),
+                ),
+            ).into())
+        }
+
+        fn assert_panic(
+            ecx: &mut InterpCx<'mir, 'tcx, Self>,
+            msg: &AssertMessage<'tcx>,
+            unwind: Option<BasicBlock>
+        ) -> InterpResult<'tcx> {
+            Err(InterpError::Unsupported(
+                UnsupportedOpInfo::Unsupported(
+                    "assert_panic".into(),
+                ),
+            ).into())
+        }
+
+        fn binary_ptr_op(
+            ecx: &InterpCx<'mir, 'tcx, Self>,
+            bin_op: BinOp,
+            left: ImmTy<'tcx, Self::PointerTag>,
+            right: ImmTy<'tcx, Self::PointerTag>
+        ) -> InterpResult<'tcx, (Scalar<Self::PointerTag>, bool, Ty<'tcx>)> {
+            Err(InterpError::Unsupported(
+                UnsupportedOpInfo::Unsupported(
+                    "binary_ptr_op".into(),
+                ),
+            ).into())
+        }
+
+        fn tag_static_base_pointer(
+            _memory_extra: &Self::MemoryExtra,
+            _id: AllocId,
+        ) -> Self::PointerTag {
+            ()
+        }
+
+        fn box_alloc(
+            ecx: &mut InterpCx<'mir, 'tcx, Self>,
+            dest: PlaceTy<'tcx, Self::PointerTag>
+        ) -> InterpResult<'tcx> {
+            Err(InterpError::Unsupported(
+                UnsupportedOpInfo::Unsupported(
+                    "box_alloc".into(),
+                ),
+            ).into())
+        }
+
+        fn init_allocation_extra<'b>(
+            memory_extra: &Self::MemoryExtra,
+            id: AllocId,
+            alloc: Cow<'b, Allocation>,
+            kind: Option<MemoryKind<Self::MemoryKinds>>
+        ) -> (Cow<'b, Allocation<Self::PointerTag, Self::AllocExtra>>, Self::PointerTag) {
+            (alloc, ())
+        }
+
+        fn stack_push(
+            ecx: &mut InterpCx<'mir, 'tcx, Self>
+        ) -> InterpResult<'tcx, Self::FrameExtra> {
+            Err(InterpError::Unsupported(
+                UnsupportedOpInfo::Unsupported(
+                    "stack_push".into(),
+                ),
+            ).into())
+        }
+
+        fn ptr_to_int(
+            _mem: &Memory<'mir, 'tcx, Self>,
+            _ptr: Pointer<Self::PointerTag>
+        ) -> InterpResult<'tcx, u64> {
+            Err(InterpError::Unsupported(
+                UnsupportedOpInfo::Unsupported(
+                    "ptr_to_int".into(),
+                ),
+            ).into())
+        }
+    }
+}
+
```

Comment:
Wow. Constants are complicated!

Reply:
Yep!  Fortunately, we were able to use rustc's const evaluation infrastructure (with this dummy `Machine` impl) to read out the discriminants and fields of evaluated constants.  Someday we may need to implement that functionality ourselves to handle transmutes in mir-verifier, but for now, this saves us a lot of work.